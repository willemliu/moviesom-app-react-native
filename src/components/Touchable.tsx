import React from 'react';
import { Platform, TouchableNativeFeedback, GestureResponderEvent, TouchableHighlight } from 'react-native';

export interface Props {
    background?: any;
    handleOnPress?: (event: GestureResponderEvent) => void;
    children?: null|undefined|JSX.Element|JSX.Element[];
}

export default class Touchable extends React.PureComponent<any, any> {
    render() {
        if (Platform.OS === 'android') {
            return (
                <TouchableNativeFeedback
                    onPress={this.props.onPress}
                    background={TouchableNativeFeedback.SelectableBackground()}
                >{this.props.children}</TouchableNativeFeedback>
            );
        } else {
            return (
                <TouchableHighlight onPress={this.props.onPress}>{this.props.children}</TouchableHighlight>
            );
        }
    }
}
