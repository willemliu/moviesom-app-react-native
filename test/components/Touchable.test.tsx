import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import Touchable from '../../src/components/Touchable';

const onPressHandler = () => {
    console.log('Pressed');
};

it('should render the correct Touchable component', () => {
    const rendered = renderer.create(<Touchable onPress={onPressHandler}><Text>Touchable component</Text></Touchable>).toJSON();
    expect(rendered).toMatchSnapshot();
});
