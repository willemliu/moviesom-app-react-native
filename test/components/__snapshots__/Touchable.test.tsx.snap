// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable should render the Android Touchable component 1`] = `
<Text
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  hitSlop={undefined}
  nativeBackgroundAndroid={
    Object {
      "attribute": "selectableItemBackground",
      "type": "ThemeAttrAndroid",
    }
  }
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID={undefined}
>
  Touchable component
</Text>
`;

exports[`Touchable should render the iOS Touchable component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    Touchable component
  </Text>
</View>
`;
